luarules/gadgets/unit_xp_range_bonus.lua

```diff
@@ -14,51 +14,53 @@ if not gadgetHandler:IsSyncedCode() then
     return false
 end
 
-
 local SetUnitWeaponState = Spring.SetUnitWeaponState
 local GetUnitExperience = Spring.GetUnitExperience
 local SetUnitMaxRange = Spring.SetUnitMaxRange
 local GetUnitDefID = Spring.GetUnitDefID
 
 local updateList = {}
+local XPDefs = {}
+
+for unitDefID, unitDef in pairs(UnitDefs) do
+	if unitDef.customParams ~= nil then
+		if unitDef.customParams.rangexpscale ~= nil then
+			XPDefs[unitDefID] = {unitDef.customParams.rangexpscale, WeaponDefs[unitDef.weapons[1].weaponDef].range}
+		end
+	end
+end
 
 function gadget:GameFrame(n)
-	for unitID in pairs(updateList) do
+	for unitID, unitDefID in pairs(updateList) do
 		local currentXP = GetUnitExperience(unitID)
         if currentXP then
-			local unitDef = UnitDefs[GetUnitDefID(unitID)]
-
-			local rangeXPScale = unitDef.customParams.rangexpscale
+			local rangeXPScale, originalRange = XPDefs[unitDefID]
```

Comment:
i think you need to to 
```suggestion
			local rangeXPScale, originalRange = XPDefs[unitDefID][1], XPDefs[unitDefID][2]
```
or alternatively
```suggestion
			local rangeXPScale, originalRange = unpack(XPDefs[unitDefID])
```

Reply:
Indeed, thanks for the correction!

Reply:
No magic numbers please, use named keys if we need to do this.

Reply:
Fixed this :)